lib/src/sai_redis_generic_get_stats.cpp

```diff
@@ -0,0 +1,220 @@
+#include "sai_redis.h"
+#include "meta/saiserialize.h"
+
+sai_status_t internal_redis_get_stats_process(
+        _In_ sai_object_type_t object_type,
+        _In_ uint32_t count,
+        _Out_ uint64_t *counter_list,
+        _In_ swss::KeyOpFieldsValuesTuple &kco)
+{
+    SWSS_LOG_ENTER();
+
+    // key:         sai_status
+    // field:       stat_id
+    // value:       stat_value
+
+    const auto &key = kfvKey(kco);
+    const auto &values = kfvFieldsValues(kco);
+
+    auto str_sai_status = key;
+
+    sai_status_t status;
+
+    sai_deserialize_status(str_sai_status, status);
+
+    if (status == SAI_STATUS_SUCCESS)
+    {
+        uint32_t i = 0;
+        for (const auto &v : values)
+        {
+            if (i >= count)
+            {
+                SWSS_LOG_ERROR("Received more values than expected");
+                status = SAI_STATUS_FAILURE;
+                break;
+            }
+
+            uint64_t value = 0;
+
+            value = stoull(fvValue(v));
+            counter_list[i] = value;
+            i++;
+        }
+    }
+
+    return status;
+}
+
+template <class T> class stat_traits {};
+
+template <>
+class stat_traits<sai_port_stat_t>
+{
+    public:
+        typedef std::string (*serialize_stat)(sai_port_stat_t);
```

Comment:
>typedef [](start = 8, length = 7)

typedef can be moved to the generic class. So only one is needed.

Reply:
It cannot. It is deferred from template parameter type, for which I define a specialization,

Reply:
I see. Thanks!